src/Database/Query.php

```diff
@@ -237,7 +237,7 @@ public function getConnection(): Connection
      */
     public function useReadRole()
     {
-        if ($this->_connection->role() !== Connection::ROLE_READ) {
+        if ($this->_connection->role() !== Connection::ROLE_READ && $this->_connection->configName()) {
```

Comment:
I think we should just enforce the connection to be a non empty string in the constructor instead. I have contemplated changing the constructor signature to `(string $name, array $options)`.

Reply:
The real problem is the config name doesn't exist in ConnectionManager.  It should fail to find a matching read and fallback to write, but it will still try to re-load that write connection. 

If users can have names in non-manager instances, we have to work around that somehow.

Reply:
Okay